prefs.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const Gio = imports.gi.Gio;
+const Gtk = imports.gi.Gtk;
+
+const ExtensionUtils = imports.misc.extensionUtils;
+const Me = ExtensionUtils.getCurrentExtension();
+
+
+function init() {
+}
+
+function buildPrefsWidget() {
+    this.settings = ExtensionUtils.getSettings(
+        'org.gnome.shell.extensions.x11gestures');
+
+    // Create a parent widget that we'll return from this function
+    let prefsWidget = new Gtk.Grid({
+        margin_top: 10,
+        margin_bottom: 10,
+        margin_start: 10,
+        margin_end: 10,
+        column_spacing: 12,
+        row_spacing: 12,
+        visible: true
+    });
+
+    // Add a simple title and add it to the prefsWidget
+    let title = new Gtk.Label({
+        label: `<b>${Me.metadata.name} Preferences</b>`,
+        halign: Gtk.Align.START,
+        use_markup: true,
+        visible: true
+    });
+    prefsWidget.attach(title, 0, 0, 2, 1);
+
+    // Create a label & switch for `show-indicator`
+    let swipeFingerLabel = new Gtk.Label({
+        label: 'Number of fingers for Swipe action binding (Extension restart needed):',
```

Comment:
Can't we avoid restarting the extension? Some users may not know how to do it.

Reply:
I didn't find out how to do it programmatically. Do you have any suggestions to recreate the configuration automatically?

Another problem strictly related here is how to clean the touchegg.config file before restart -expecially if touché or some other tools rewrote the file. We should also inform the user of this problem.
Furthermore, in this case I noticed that 3 finger gestures override the 4 finger ones. So if you restart the extension without cleaning the touchegg.config file you won't notice any difference.

I thought this case should be handled in a separate PR, started looking on how to parse and serialize XML in gnome extensions and discovered how hard it can be, as you said 😄